File: java/src/main/java/io/fivexer/sdk/model/WorkerTaskDetail.java
package io.fivexer.sdk.model;

import java.util.List;
import java.util.Map;

/**
 * Rich detail for a task assigned to the authenticated worker (portal surface only).
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerTaskDetail {
    private String id;
    private String status;
    private List<String> tags;
    private Double priority;
    private String title;
    private String description;
    private Map<String, Object> context;
    private List<TaskReference> references;
    private Long createdAt;

    public String getId() { return id; }
    public String getStatus() { return status; }
    public List<String> getTags() { return tags; }
    public Double getPriority() { return priority; }
    public String getTitle() { return title; }
    public String getDescription() { return description; }
    public Map<String, Object> getContext() { return context; }
    public List<TaskReference> getReferences() { return references; }
    public Long getCreatedAt() { return createdAt; }
}

File: php/src/Model/WorkerTaskDetail.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * Rich detail for a task assigned to the authenticated worker.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerTaskDetail
{
    public function __construct(
        public readonly string $id,
        public readonly string $status,
        /** @var list<string> */
        public readonly array $tags,
        public readonly ?float $priority,
        public readonly ?string $title,
        public readonly ?string $description,
        /** @var array<string, mixed>|null */
        public readonly ?array $context,
        /** @var list<TaskReference>|null */
        public readonly ?array $references,
        public readonly ?int $createdAt,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            status: (string) ($data['status'] ?? ''),
            tags: \array_map('strval', $data['tags'] ?? []),
            priority: isset($data['priority']) ? (float) $data['priority'] : null,
            title: isset($data['title']) ? (string) $data['title'] : null,
            description: isset($data['description']) ? (string) $data['description'] : null,
            context: $data['context'] ?? null,
            references: Json::parseEachOrNull($data, 'references', [TaskReference::class, 'fromArray']),
            createdAt: isset($data['createdAt']) ? (int) $data['createdAt'] : null,
        );
    }
}
